src/SFA.DAS.ApprenticeCommitments/Application/Commands/RegistrationFirstSeenCommand/RegistrationFirstSeenCommandHandler.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+using MediatR;
+using SFA.DAS.ApprenticeCommitments.Data;
+
+namespace SFA.DAS.ApprenticeCommitments.Application.Commands.RegistrationFirstSeenCommand
+{
+    public class RegistrationFirstSeenCommandHandler : IRequestHandler<RegistrationFirstSeenCommand>
+    {
+        private readonly IRegistrationContext _registrations;
+
+        public RegistrationFirstSeenCommandHandler(IRegistrationContext registrations)
+        {
+            _registrations = registrations;
+        }
+
+        public async Task<Unit> Handle(Commands.RegistrationFirstSeenCommand.RegistrationFirstSeenCommand command, CancellationToken _)
+        {
+            var registration = await _registrations.GetById(command.ApprenticeId);
+            registration.SetFirstViewedOn(command.SeenOn);
```

Comment:
I wonder if it may be worth making the method name more of the action that has happened?  

The instruction to update a property feels a bit like it's communicating the mechanics of what to do, and is factually incorrect the second time round.

```suggestion
            registration.ViewedByUser(command.SeenOn);
```

Reply:
agreed, better name